From 49d0b0e666fc0d858d81974c05539963faf08217 Mon Sep 17 00:00:00 2001
From: Colin Ian King <colin.i.king@gmail.com>
Date: Sun, 26 May 2024 16:08:20 +0100
Subject: [PATCH] Makefile.config: add libgmp when checking libmpfr

mpfr requires gmp, so add the library when build checking
for mpfr. Also update link order for static linking

Closes: https://github.com/ColinIanKing/stress-ng/issues/390

Signed-off-by: Colin Ian King <colin.i.king@gmail.com>
---
 Makefile.config | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/Makefile.config b/Makefile.config
index 6fa76df33..de789b7da 100644
--- a/Makefile.config
+++ b/Makefile.config
@@ -58,8 +58,8 @@ LIB_NSL := -lnsl
 
 LIB_ORDER := $(LIB_ACL) $(LIB_AIO) $(LIB_APPARMOR) $(LIB_ATOMIC) $(LIB_BSD) \
 	$(LIB_CRYPT) $(LIB_DL) $(LIB_IPSEC_MB) $(LIB_JPEG) $(LIB_JUDY) \
-	$(LIB_KMOD) $(LIB_EGL) $(LIB_GLES2) $(LIB_GMP) $(LIB_GBM) $(LIB_MD) \
-	$(LIB_MPFR) $(LIB_SCTP) $(LIB_XXHASH) $(LIB_Z) $(LIB_RT) \
+	$(LIB_KMOD) $(LIB_EGL) $(LIB_GLES2) $(LIB_MPFR) $(LIB_GMP) $(LIB_GBM) $(LIB_MD) \
+	$(LIB_SCTP) $(LIB_XXHASH) $(LIB_Z) $(LIB_RT) \
 	$(LIB_PTHREAD) $(LIB_MATH) $(LIB_NETWORK) $(LIB_SOCKET) $(LIB_NSL) $(LIB_C)
 
 ifeq ($(shell $(CC) -v 2>&1 | grep 'gcc version' | grep -v 'icc' | wc -l),1)
@@ -334,7 +334,10 @@ all: libraries headers cpufeatures types symbols functions
 	@cat $(CONFIGS)/* | grep -v "#" | grep CONFIG_LDFLAGS | sed '/^$$/d' | sort | uniq > config.sort
 	@rm -f config
 	@for I in $(LIB_ORDER); do \
+		echo ""; \
+		echo $$I; \
 		grep "CONFIG_LDFLAGS += $$I\$$" config.sort >> config; \
+		cat config; \
 		echo "" >> config.sort; \
 	done
 	@rm -f config.sort $(CONFIGS)/HAVE_LDFLAGS_TMP
@@ -406,7 +409,7 @@ LIB_MD:
 	$(call check,test-libmd,HAVE_LIB_MD,$(LIB_MD),$(LIB_MD))
 
 LIB_MPFR:
-	$(call check,test-libmpfr,HAVE_LIB_MPFR,$(LIB_MPFR),$(LIB_MPFR))
+	$(call check,test-libmpfr,HAVE_LIB_MPFR,$(LIB_MPFR),$(LIB_MPFR) $(LIB_GMP))
 
 LIB_PTHREAD:
 	$(call check,test-libpthread,HAVE_LIB_PTHREAD,$(LIB_PTHREAD),$(LIB_PTHREAD))
